.github/workflows/release.yml

```diff
@@ -0,0 +1,121 @@
+---
+name: release
+
+on: push
```

Comment:
This works for testing, since you can just push to a branch, but...
- Do you want to run this on all pushes or just tags that match the pattern? Or pushes to branches that match a pattern?
- Do you want to add a workflow_dispatch trigger? Might be nice for debugging.

Reply:
I want to attempt to build the installable package (the first job below) on push so that if it breaks we know immediately and can fix it while we still have context.

The filter which prevents any of the actual publication from happening is in the first of the subsequent publishing jobs.  If we want to do filtering in the `on:` step then we need to create a separate `build-package.yml` workflow that just attempts to create the package as part of the CI, and duplicate that logic in here, which seemed less clear & concise.

Reply:
As far as the `workflow_dispatch` I don't think it will help with the testing, since so much of what happens downatream is tied to having the version-compatible github ref to work with.  We use `setuptools_scm` to automatically generate a software version from the tags, and if we don't do that we've got a garbled mess of git stuff as the "version" which PyPI rejects (not sure what GitHub thinks about it).

Mostly what I need to test here is that these actions are hooked up to the tagging and we have all the authentication setup. I've used these actions in `ferc-xbrl-extractor` before so hopefully they'll work here as well..